chain/client/src/stateless_validation/chunk_validator.rs

```diff
@@ -60,8 +60,15 @@ impl ChunkValidator {
         epoch_manager: Arc<dyn EpochManagerAdapter>,
         network_sender: Sender<PeerManagerMessageRequest>,
         runtime_adapter: Arc<dyn RuntimeAdapter>,
+        chunk_endorsement_tracker: Arc<ChunkEndorsementTracker>,
```

Comment:
I have a bit of a mixed feelings about this, seem a bit strange to have the validator have access to the tracker. That being said I like it more than what we have now so you have my approval.

Reply:
Figuring out which module depends on what is a HUGE headache imo.... I spent a good amount of time thinking about this and I felt this is probably the best approach.

The other idea I had was to create two separate traits, one related to chunk endorsement tracking and one related to validation, and have a single module implement both traits. That way there wouldn't technically be any dependency between these as they are no longer modules, just traits....